### PR TITLE
Joystick input

### PR DESCRIPTION
**Pull request by ryan-feng**
Code to read joystick input and store x and y values and attempts to send information in a UDP packet.

### CODE REVIEW

**Comment by twpryor** [comment]
Hi Ryan,

Is this code for the Xbox controller or the joysticks? If it's for the Xbox controller, could you put in a file in the same directory named xboxController.py

**Comment by ryan-feng** [comment]
Hi Tommy,

The JoystickInputTest.py file I uploaded in the first joystick commit is for the joysticks. The controllers technically work but only the left stick has any functionality.

I uploaded a few minutes ago a file called logitechController.py that was modified specifically for input on the logitech controller. This one should allow for input on both of the controller sticks. What it is supposed to do is add the inputs from both of the sticks together and if the value is outside of the range [0, 255] it sets it to the closer bound. In this way, pushing the two sticks in the same direction is the same as pushing the joystick and the thrust button at the same time. I can modify it to handle the two inputs differently as preferred, wasn't sure how it was preferred to combine them when I wrote it.

Hopefully that made sense and if there is anything I should clarify or edit in the code just let me know. Thanks!

**Comment by ryan-feng** [comment]
I modified the code to keep the inputs of the two sticks separate for separate outputs. Right now the thrust button is the 1 button which is kind of inconvenient. I can swap it for another one but I would need the controller to figure out which button id numbers are which

File: JoystickInputTest.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            p.join(100)
+            if(p.is_alive()):
+                print("Disconnect... Attempting to fix.")
+                p.terminate()
+                p.join()'''
+
+            pygame.time.wait(100)
+
+    except KeyboardInterrupt:
+        pygame.quit()
```
Comment (twpryor) [inline JoystickInputTest.py]: Be sure to always add an extra newline at the end of every Python script that you write.

File: test.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+
+code
```
Comment (twpryor) [inline test.txt]: Delete this file.

File: JoystickControllers/logitechController.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            p.join(100)
+            if(p.is_alive()):
+                print("Disconnect... Attempting to fix.")
+                p.terminate()
+                p.join()'''
+
+            pygame.time.wait(100)
+
+    except KeyboardInterrupt:
+        pygame.quit()
```
Comment (twpryor) [inline JoystickControllers/logitechController.py]: Add newline.

File: JoystickControllers/JoystickLogitech.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+            p.join(100)
+            if(p.is_alive()):
+                print("Disconnect... Attempting to fix.")
+                p.terminate()
+                p.join()'''
+
+            pygame.time.wait(100)
+
+    except KeyboardInterrupt:
+        pygame.quit()
```
Comment (twpryor) [inline JoystickControllers/JoystickLogitech.py]: Newline

**Comment by twpryor** [comment]
Be sure to clean up the code by deleting any unused functions and variables like ```arm[]``` and ```hand[]```. We are only driving the motors for the prototype. Could you also switch your pull request to request a merge into ```huskyroboticsteam:prototype``` instead?

**Pull request closed without merging**